flink-ml-lib/src/test/java/org/apache/flink/ml/feature/VectorSlicerTest.java

```diff
@@ -0,0 +1,229 @@
+        vectorSlicer.setInputCol("vec").setOutputCol("sliceVec").setIndices(0, 1, 2);
+        assertEquals("vec", vectorSlicer.getInputCol());
+        assertEquals("sliceVec", vectorSlicer.getOutputCol());
+        assertArrayEquals(new Integer[] {0, 1, 2}, vectorSlicer.getIndices());
+    }
+
+    @Test
+    public void testSaveLoadAndTransform() throws Exception {
+        VectorSlicer vectorSlicer =
+                new VectorSlicer().setInputCol("vec").setOutputCol("sliceVec").setIndices(0, 1, 2);
+        VectorSlicer loadedVectorSlicer =
+                TestUtils.saveAndReload(
+                        tEnv, vectorSlicer, TEMPORARY_FOLDER.newFolder().getAbsolutePath());
+        Table output = loadedVectorSlicer.transform(inputDataTable)[0];
+        verifyOutputResult(output, loadedVectorSlicer.getOutputCol(), false);
+    }
+
+    @Test
+    public void testEmptyIndices() {
+        try {
+            VectorSlicer vectorSlicer =
+                    new VectorSlicer().setInputCol("vec").setOutputCol("sliceVec").setIndices();
+            vectorSlicer.transform(inputDataTable);
+            Assert.fail("Expected IllegalArgumentException");
+        } catch (Exception e) {
+            assertEquals("Parameter indices is given an invalid value {}", e.getMessage());
+            assertEquals(IllegalArgumentException.class, ((Throwable) e).getClass());
+        }
+    }
+
+    @Test
+    public void testIndicesLargerThanVectorSize() {
+        try {
+            VectorSlicer vectorSlicer =
+                    new VectorSlicer()
+                            .setInputCol("vec")
+                            .setOutputCol("sliceVec")
+                            .setIndices(1, 2, 10);
+            Table output = vectorSlicer.transform(inputDataTable)[0];
+            DataStream<Row> dataStream = tEnv.toDataStream(output);
+            IteratorUtils.toList(dataStream.executeAndCollect());
+            Assert.fail("Expected RuntimeException");
+        } catch (Exception e) {
+            assertEquals(
+                    "Index value 10 is greater than vector size:6",
+                    ExceptionUtils.getRootCause(e).getMessage());
+            assertEquals(IllegalArgumentException.class, ExceptionUtils.getRootCause(e).getClass());
+        }
+    }
+
+    @Test
+    public void testIndicesSmallerThanZero() {
+        try {
+            new VectorSlicer().setInputCol("vec").setOutputCol("sliceVec").setIndices(1, -2);
+            Assert.fail("Expected IllegalArgumentException");
+        } catch (Exception e) {
+            assertEquals("Parameter indices is given an invalid value {1,-2}", e.getMessage());
+            assertEquals(IllegalArgumentException.class, e.getClass());
+        }
+    }
+
+    @Test
+    public void testDuplicateIndices() {
+        try {
+            new VectorSlicer().setInputCol("vec").setOutputCol("sliceVec").setIndices(1, 1, 3);
+            Assert.fail("Expected IllegalArgumentException");
```

Comment:
what about replace this line with `fail();`? We are assuming that the code never gets here. And it is not a IllegalArgumentException.

Same for other `Assert.fails`.